test/helpers/kubectl.go

```diff
@@ -1737,49 +1737,72 @@ func (kub *Kubectl) GetCiliumPodOnNode(namespace string, node string) (string, e
 	return res.Output().String(), nil
 }
 
+func (kub *Kubectl) ciliumPreFlightCheck() error {
+	switch GetCurrentIntegration() {
+	case CIIntegrationFlannel:
+	default:
+		err := kub.ciliumStatusPreFlightCheck()
+		if err != nil {
+			return fmt.Errorf("status is unhealthy: %s", err)
+		}
+	}
+
+	err := kub.ciliumControllersPreFlightCheck()
+	if err != nil {
+		return fmt.Errorf("controllers are failing: %s", err)
+	}
+
+	switch GetCurrentIntegration() {
+	case CIIntegrationFlannel:
+	default:
+		err = kub.ciliumHealthPreFlightCheck()
+		if err != nil {
+			return fmt.Errorf("connectivty health is failing: %s", err)
+		}
+	}
+	err = kub.fillServiceCache()
+	if err != nil {
+		return fmt.Errorf("unable to fill service cache: %s", err)
+	}
+	err = kub.ciliumServicePreFlightCheck()
+	if err != nil {
+		return fmt.Errorf("cilium services are not set up correctly: %s", err)
+	}
+	err = kub.servicePreFlightCheck("kubernetes", "default")
+	if err != nil {
+		return fmt.Errorf("kubernetes service is not ready: %s", err)
+	}
+
+	return nil
+}
+
 // CiliumPreFlightCheck specify that it checks that various subsystems within
 // Cilium are in a good state. If one of the multiple preflight fails it'll
 // return an error.
 func (kub *Kubectl) CiliumPreFlightCheck() error {
 	ginkgoext.By("Performing Cilium preflight check")
 	// Doing this withTimeout because the Status can be ready, but the other
 	// nodes cannot be show up yet, and the cilium-health can fail as a false positive.
-	var err error
-	body := func() bool {
-		switch GetCurrentIntegration() {
-		case CIIntegrationFlannel:
-		default:
-			err = kub.CiliumStatusPreFlightCheck()
-			if err != nil {
-				return false
-			}
-		}
-		err = kub.CiliumControllersPreFlightCheck()
-		if err != nil {
-			return false
-		}
+	var (
+		err                 error
+		lastError           string
+		consecutiveFailures int
+	)
 
-		switch GetCurrentIntegration() {
-		case CIIntegrationFlannel:
-		default:
-			err = kub.CiliumHealthPreFlightCheck()
-			if err != nil {
-				return false
+	body := func() bool {
+		if err := kub.ciliumPreFlightCheck(); err != nil {
+			newError := err.Error()
+			if lastError != newError || consecutiveFailures >= 5 {
+				ginkgoext.By("Cilium is not ready yet: %s", newError)
```

Comment:
I would not add this By here, but I understand why  you have it.

Reply:
@eloycoto Where would you add it?

Reply:
For me seen 5 times that Cilium is not ready does not really add a real value,
it's always the same message with an error, I would make something like this
with this function. 

```
func (kub *Kubectl) CiliumPreFlightCheck() error {
	ginkgoext.By("Performing Cilium preflight check")
	// Doing this withTimeout because the Status can be ready, but the other
	// nodes cannot be show up yet, and the cilium-health can fail as a false positive.
	var (
        errors []error
	)

	body := func() bool {
        err := kub.ciliumPreFlightCheck();
        if err != nil {
            errors = append(errors, err)
            return false
        }
        return true
    }

	timeoutErr := WithTimeout(body, "PreflightCheck failed", &TimeoutConfig{Timeout: HelperTimeout})
	if timeoutErr != nil {
        ginkgoext.By("Cilium  preflight is not ready after timeout, error stages:")
        for _,item := range errors {
            ginkgoext.GinkoPrint("%s", item)
        }
		return fmt.Errorf("CiliumPreFlightCheck error: %s: %s", timeoutErr, err)
	}
	return nil
}
```

You make the function to log errors always, with this way the errors will only
seen when things fail. But this is only personal style. 

Best, 
Eloy